_posts/2016-06-10-multifidelity-emulation.markdown

```diff
@@ -2,288 +2,97 @@
+to as high-fidelity observations, whereas approximations are denoted as being low-fidelity.
+These low-fidelity  observations are then systemically combined with the more accurate (but limited) observations in order to predict 
+the high-fidelity output more effectively. 
+
+Emukit offers implementation of a selection Gaussian process multi-fidelity models that can also be combined with other 
+outer-loop applications. To work with an specific example we start loading a problem with two fidelities in which their relationship 
+between is given by $$f_{high}(x) = (x-\sqrt{2})f^2_{low}$$. You can check 
+[[2](#references-on-multi-fidelity-gaussian-processes)] for a graphical representation of this problem. We will use the interval [0,1]
+as the input domain in our experiment.
+
+```python
+from emukit.emukit.test_functions.non_linear_sin import nonlinear_sin_high, nonlinear_sin_low
+n_low_fidelity_points = 50
+
+x_train_l = nonlinear_sin_high(np.linspace(0, 1, n_low_fidelity_points)[:, None])
+x_train_h = nonlinear_sin_low(x_train_l[::4, :])
+
+X_train, Y_train = convert_xy_lists_to_arrays([x_train_l, x_train_h], [y_train_l, y_train_h])
+``` 
+ 
+With this code we have just collected some samples in both fidelities and formatted the data in a way that can be understood by Emukit. 
+Normally, high fidelities are more expensive to sample than low fidelities so we have reflected that in the 
+data collection approach.  
+ 
+Now that we have the data, it is time to show how to use Emukit to train a Gaussian process *linear* multi-fidelity model as proposed 
+in [[1]](#references-on-multi-fidelity-gaussian-processes). This models assumes that
+the high fidelity is a linear combination of the low fidelity and a *delta* term that captures the difference between the two. 
+We just need to specify the kernels for the two fidelities and create the model. [GPy](https://github.com/SheffieldML/GPy) is used as 
+the modelling framework in this example.
+
+```python
+import GPy
+from emukit.multi_fidelity.kernels import LinearMultiFidelityKernel
+from emukit.multi_fidelity.models import GPyLinearMultiFidelityModel
+
+num_fidelities = 2
+kernels = [GPy.kern.RBF(1), GPy.kern.RBF(1)]
+linear_mf_kernel = LinearMultiFidelityKernel(kernels)
+gpy_linear_mf_model = GPyLinearMultiFidelityModel(X_train, Y_train, lin_mf_kernel, n_fidelities = 2)
+```
+
+We have created the model. The last step is to wrap it (in case we want to use it in a loop like Bayesian optimization) and to train it. 
+As the evaluations of the fidelities are exact, we first set the noise parameters to zero by doing
+
+```python
+gpy_linear_mf_model.mixed_noise.Gaussian_noise.fix(0)
+gpy_linear_mf_model.mixed_noise.Gaussian_noise_1.fix(0)
+```
+and complete the training by writing 
+
+```python
+from emukit.model_wrappers.gpy_model_wrappers import GPyMultiOutputWrapper
+
+linear_mf_model =  GPyMultiOutputWrapper(gpy_linear_mf_model, num_fidelities, n_optimization_restarts = 5)
```

Comment:
Why would we need a wrapper? You could just call optimize right away. There is no need to use the wrapper unless the model is used for some Emukit loop.

You could however just call `optimize` on the model here, and then separately mention that one could use `GPyMultiOutputWrapper` if they wanted to use these models with Emukit

Reply:
Good point, done.